# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLIpdl.h
# Commit: 6149f5b15c5e
# Full Hash: 6149f5b15c5e278ef15ea36fa896fa8c146ebdca
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2021-01-14 08:32:45
# Regressor Bug: 1640607
# File Overlap Count: 1
# Description:
#   Bug 1640607 - Send SurfaceDescriptors for GPU blitting for video-to-webgl. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101061
# ==============================================================================

diff -r 29fd3a6a7e27 -r 6149f5b15c5e dom/canvas/WebGLIpdl.h
--- a/dom/canvas/WebGLIpdl.h	Wed Jan 13 20:14:54 2021 +0000
+++ b/dom/canvas/WebGLIpdl.h	Wed Jan 13 21:49:40 2021 +0000
@@ -226,6 +226,10 @@
 
 // -
 
+template <typename T>
+struct ParamTraits<mozilla::webgl::EnumMask<T>> final
+    : public PlainOldDataSerializer<mozilla::webgl::EnumMask<T>> {};
+
 template <>
 struct ParamTraits<mozilla::webgl::InitContextResult> final {
   using T = mozilla::webgl::InitContextResult;
@@ -234,13 +238,15 @@
     WriteParam(msg, in.error);
     WriteParam(msg, in.options);
     WriteParam(msg, in.limits);
+    WriteParam(msg, in.uploadableSdTypes);
   }
 
   static bool Read(const Message* const msg, PickleIterator* const itr,
                    T* const out) {
     return ReadParam(msg, itr, &out->error) &&
            ReadParam(msg, itr, &out->options) &&
-           ReadParam(msg, itr, &out->limits);
+           ReadParam(msg, itr, &out->limits) &&
+           ReadParam(msg, itr, &out->uploadableSdTypes);
   }
 };
 